app/src/main/java/org/mozilla/fenix/nimbus/controller/NimbusBranchesController.kt

```diff
@@ -4,28 +4,59 @@
 
 package org.mozilla.fenix.nimbus.controller
 
+import android.content.Context
 import mozilla.components.service.nimbus.NimbusApi
 import mozilla.components.service.nimbus.ui.NimbusBranchesAdapterDelegate
 import org.mozilla.experiments.nimbus.Branch
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.FenixSnackbar
+import org.mozilla.fenix.components.metrics.MetricServiceType
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.getRootView
+import org.mozilla.fenix.ext.settings
 import org.mozilla.fenix.nimbus.NimbusBranchesAction
 import org.mozilla.fenix.nimbus.NimbusBranchesStore
 
 /**
  * [NimbusBranchesFragment] controller. This implements [NimbusBranchesAdapterDelegate] to handle
- * interactions with a Nimbus branch item.
+ * interactions with a Nimbus branch.
  *
  * @param nimbusBranchesStore An instance of [NimbusBranchesStore] for dispatching
  * [NimbusBranchesAction]s.
  * @param experiments An instance of [NimbusApi] for interacting with the Nimbus experiments.
  * @param experimentId The string experiment-id or "slug" for a Nimbus experiment.
  */
 class NimbusBranchesController(
+    private val context: Context,
     private val nimbusBranchesStore: NimbusBranchesStore,
     private val experiments: NimbusApi,
     private val experimentId: String
 ) : NimbusBranchesAdapterDelegate {
 
     override fun onBranchItemClicked(branch: Branch) {
+        val telemetryEnabled = context.settings().isTelemetryEnabled
+        val experimentsEnabled = context.settings().isExperimentationEnabled
+
+        if (!telemetryEnabled && !experimentsEnabled) {
+            updateOptInState(branch)
+            context.components.analytics.metrics.stop(MetricServiceType.Data)
```

Comment:
If `isTelemetryEnabled` is false, shouldn't `DataChoicesFragment` already called `context.components.analytics.metrics.stop(MetricServiceType.Data)`?

Reply:
In the case of opt **IN**, we need to call this again even if the preference has been disabled already